starboard/nplb/posix_compliance/posix_thread_attr_test.cc

```diff
@@ -23,37 +24,183 @@ constexpr int kStackSize = 4 * 1024 * 1024;
+
+  // Verify retrieved values.
+  EXPECT_EQ(ret_stack_addr, set_stack_addr)
+      << "Retrieved stack address does not match set address.";
+
+  EXPECT_GE(ret_stack_size, kStackSize)
+      << "Retrieved stack size (" << ret_stack_size
+      << ") is less than set size (" << kStackSize << ").";
+
+  ret = pthread_attr_destroy(&attr);
+  EXPECT_EQ(ret, 0) << "pthread_attr_destroy failed: " << strerror(ret);
+}
+
+// Test for getting and setting the contention scope attribute.
+TEST(PosixThreadAttrTest, ScopeAttr) {
+  pthread_attr_t attr;
+  int ret = pthread_attr_init(&attr);
+  ASSERT_EQ(ret, 0) << "pthread_attr_init failed: " << strerror(ret);
+
+  int scope = -1;  // Initialize to an invalid value.
+
+  // --- Test PTHREAD_SCOPE_PROCESS ---
+  const int process_scope = PTHREAD_SCOPE_PROCESS;
+  ret = pthread_attr_setscope(&attr, process_scope);
+
+  if (ret == ENOTSUP) {
+    // PTHREAD_SCOPE_PROCESS might not be supported (common on Linux).
+    SB_DLOG(INFO) << "Setting scope to PTHREAD_SCOPE_PROCESS is not supported.";
+  } else {
+    EXPECT_EQ(ret, 0)
+        << "pthread_attr_setscope (to PROCESS) failed unexpectedly: "
+        << strerror(ret);
+
+    // If setting succeeded, get it back and verify.
+    if (ret == 0) {
+      scope = -1;  // Reset
+      ret = pthread_attr_getscope(&attr, &scope);
+      EXPECT_EQ(ret, 0) << "pthread_attr_getscope (after set PROCESS) failed: "
+                        << strerror(ret);
+      if (ret == 0) {
+        EXPECT_EQ(scope, process_scope);
+      }
+    }
+  }
+
+  // --- Test PTHREAD_SCOPE_SYSTEM ---
+  const int system_scope = PTHREAD_SCOPE_SYSTEM;
+  ret = pthread_attr_setscope(&attr, system_scope);
+  EXPECT_EQ(ret, 0) << "pthread_attr_setscope (to SYSTEM) failed: "
+                    << strerror(ret);
+
+  // If setting succeeded, get it back and verify.
+  if (ret == 0) {
+    scope = -1;  // Reset
+    ret = pthread_attr_getscope(&attr, &scope);
+    EXPECT_EQ(ret, 0) << "pthread_attr_getscope (after set SYSTEM) failed: "
+                      << strerror(ret);
+    if (ret == 0) {
+      EXPECT_EQ(scope, system_scope);
+    }
+  }
+
+  ret = pthread_attr_destroy(&attr);
+  EXPECT_EQ(ret, 0) << "pthread_attr_destroy failed: " << strerror(ret);
+}
+
+// Test for getting and setting the scheduling policy attribute.
+TEST(PosixThreadAttrTest, SchedPolicyAttr) {
+  pthread_attr_t attr;
+  int ret = pthread_attr_init(&attr);
+  ASSERT_EQ(ret, 0) << "pthread_attr_init failed: " << strerror(ret);
+
+  int policy = -1;  // Initialize to an invalid value.
+
+  // Helper lambda to test setting and getting a specific policy.
+  auto test_policy = [&](int policy_to_set, const char* policy_name) {
+    const int set_ret = pthread_attr_setschedpolicy(&attr, policy_to_set);
+    EXPECT_EQ(set_ret, 0) << "pthread_attr_setschedpolicy (to " << policy_name
+                          << ") failed unexpectedly: " << strerror(set_ret);
+    if (set_ret == 0) {
```

Comment:
nit: Prefer early return, i.e. instead of
```
  if (this) {
    if (that) {
    }
  }
``` 
rather have
```
  if (!this) {
    return;
  }
  if (that) {
  }  
``` 

In this case:
```
    if (set_ret != 0) {  // Maybe set_ret == kError
      return;
    }
```